waffle-mock-contract/src/index.ts

```diff
@@ -15,7 +16,28 @@ export interface MockContract extends Contract {
   staticcall (contract: Contract, functionName: string, ...params: any[]): Promise<any>;
 }
 
-async function deploy(signer: Signer) {
+async function deploy(signer: Signer, address?: string) {
```

Comment:
Consider adding an option allowing to override the existing address. So the signature will look something like 
```ts
type DeployOptions = {
  address?: string
} | {
  address: string,
  overrideExsiting: boolean
}

async function deploy(signer: Signer, options?: DeployOptions = {}) {
```